wdl-lint/src/rules/mixed_indentation.rs

```diff
@@ -0,0 +1,189 @@
+            Self::Tabs => write!(f, "tabs"),
+        }
+    }
+}
+
+/// Creates a "mixed indentation" diagnostic.
+fn mixed_indentation(span: Span, kind: IndentationKind) -> Diagnostic {
+    let anti_kind = match kind {
+        IndentationKind::Spaces => IndentationKind::Tabs,
+        IndentationKind::Tabs => IndentationKind::Spaces,
+    };
+
+    let fix_message = match kind {
+        IndentationKind::Spaces => "use spaces consistently throughout the document",
+        IndentationKind::Tabs => "use tabs consistently throughout the document",
+    };
+
+    Diagnostic::note("mixed indentation in document")
+        .with_rule(ID)
+        .with_label(
+            format!(
+                "indented with {kind} until this {anti}",
+                anti = match kind {
+                    IndentationKind::Spaces => "tab",
+                    IndentationKind::Tabs => "space",
+                }
+            ),
+            span,
+        )
+        .with_fix(fix_message)
+}
+
+/// Rule that checks for mixed indentation (tabs and spaces) throughout a
+/// document, excluding command sections which are handled by wdl-analysis.
+/// This rule helps ensure consistent indentation style for better readability.
+#[derive(Default, Debug, Clone)]
+pub struct MixedIndentationRule;
+
+impl Rule for MixedIndentationRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Ensures consistent indentation (no mixed spaces/tabs) throughout the document, excluding \
+         command sections which are handled by the analysis `CommandMixedIndentation` rule."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Mixing tabs and spaces in non-command sections reduces readability and can lead to \
+         inconsistent rendering depending on editor settings."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Clarity, Tag::Correctness, Tag::Spacing])
+    }
+
+    fn exceptable_nodes(&self) -> Option<&'static [SyntaxKind]> {
+        None
+    }
+
+    fn related_rules(&self) -> &[&'static str] {
+        &[]
+    }
+}
+
+impl Visitor for MixedIndentationRule {
+    fn reset(&mut self) {
+        *self = Default::default();
+    }
+
+    fn document(
+        &mut self,
+        diagnostics: &mut Diagnostics,
+        reason: VisitReason,
+        doc: &Document,
+        _version: SupportedVersion,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        let mut first_style = None;
+        let mut mixed_span = None;
+        let text = doc.root().inner().text().to_string();
+
+        // Collect all command section spans to avoid firing for them
+        let command_spans: Vec<Span> = doc
+            .root()
+            .inner()
+            .descendants()
+            .filter(|node| node.kind() == SyntaxKind::CommandSectionNode)
+            .map(|node| node.text_range().into())
+            .collect();
```

Comment:
this can be made more efficient. Instead of finding the command spans within entry of the document, you should store the command spans as they are encountered during normal AST visitation (i.e. impl `command_section` for the `Visitor` and store the spans in the rule struct) and then do the mixed indentation check at document `Exit`.